### PR TITLE
Bulk Analysis Feature

### PR DESCRIPTION
**Pull request by beesaycheese**
HotKey `B` and the menu item bulk analysis button will invoke `bulkAnalysis()` method which goes through the entire assetList to get inference for each of the asset.

Under the Advance Settings, the User is able to choose if they want the bulk analysis to include `Image Only`, `Video Only` or `Image and Video`

image

Includes end bulk analysis and end video prediction. Does not work for ending image prediction

### CODE REVIEW

File: src/app/src/components/annotations/annotator.tsx
Comment (ghost) [inline src/app/src/components/annotations/annotator.tsx]: Not sure whether you could refactor `bulkAnalysis` and `getInference`. The way I see it, they have very similar code blocks

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/app/src/components/annotations/annotator.tsx
Comment (beesaycheese) [reply src/app/src/components/annotations/annotator.tsx]: ok will make the change after the cachelist PR is merged!

**Comment by beesaycheese** [comment]
Do not merge this yet, currently facing backend backlogs due to the new socket implementation. Might have new changes for this PR.

File: src/app/src/components/annotations/annotator.tsx
Comment (beesaycheese) [reply src/app/src/components/annotations/annotator.tsx]: @virtuoso98 ZhaoYuan, I've made the requested changes and have refectored the code such that getInference() is now a general method called by bulkAnalysis() and singleAnalysis()

**Comment by beesaycheese** [comment]
PR ready to be re-reviewed

File: src/app/src/components/annotations/annotator.tsx
Comment (virtuoso98) [inline src/app/src/components/annotations/annotator.tsx:628]: Looks good to me! Correct me if I'm wrong but I'm assuming that `await` will pause the execution of line 601 until 600 completes with response?

**Review by virtuoso98 — APPROVED** [review]

File: src/app/src/components/annotations/annotator.tsx
Comment (virtuoso98) [reply src/app/src/components/annotations/annotator.tsx]: @beesaycheese , thanks for the changes! Looks good to me

File: src/app/src/components/annotations/annotator.tsx
Comment (beesaycheese) [reply src/app/src/components/annotations/annotator.tsx:628]: yeap

**Pull request merged**
